### ISSUE TITLE
Make a custom loss function

### ISSUE THREAD

**Issue by MRrollingJerry** [keras-team/keras#10750]
Hi, I'm tying to apply a custom loss function which fits my model.
The labels are binary array of shape (5,), and the position of 1 in the array matters. for example [1,0,0,0,0] is y_true. y_pred as [0,0,1,0,0] will have a higher loss than y_pred as [0,0,0,0,1] have.

The naïve function is below:

    from keras import backend as K
    def tf_loss_function(y_true, y_pred):
        y_pred = K.clip(y_pred, _Epsilon, 1.0-_Epsilon)
        true = 0
        pred = 0
        tf_session = K.get_session()
        y_true = y_true.eval(tf_session())
        y_pred = y_pred.eval(tf_session())
        shape = y_true.shape

        for row in range(shape[0]):
            for elem in range(shape[1]):
                true += y_true[elem] * elem
                pred += y_true[elem] * elem
        return ((true - pred) ** 2)/shape[0]

I know this function is completely wrong, but the idea is there. How can I write a function which I can directly fit into the model.compile()?

**Comment by tomastheod-ITI** [comment]
@MRrollingJerry `K.argmax(y_pred)` gives you the position of the highest value. Maybe you can use this to construct your loss. For
```
y_pred = K.variable([[1, 0, 0, 0, 0],
                     [0, 0, 1, 0, 0],
                     [0, 0, 0, 0, 1],
                     [0, 0, 0, 0, 0]])
```
this method returns `array([0., 2., 4., 0.], dtype=float32)` when evaluated. Beware of the behaviour for the last case!
